keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -85,11 +88,15 @@ open class SqlResourceRepository(
    * Constructs a resource object from its database representation
    */
   private fun constructResource(kind: String, metadata: String, spec: String) =
-    Resource(
-      parseKind(kind),
-      objectMapper.readValue<Map<String, Any?>>(metadata).asResourceMetadata(),
-      objectMapper.readValue(spec, resourceTypeIdentifier.identify(parseKind(kind)))
-    )
+    specMigrators
+      .migrate(parseKind(kind), objectMapper.readValue(spec, resourceTypeIdentifier.identify(parseKind(kind))))
```

Comment:
I hadn't paid too much attention to `ResourceTypeIdentifier` before, and find this a bit hard to read. Does it define the canonical kind this repository knows how to handle? If that's the case, could we perhaps rename the property for clarity? Either way, it might be useful to add a line comment describing it above.

Reply:
It's really just a function `ResourceKind -> Class<out ResourceSpec>` in other words, figure out what spec is the implementation for this kind. I feel like I can probably short circuit this entirely using the `SupportedKind` type, tho.

Reply:
Never mind. I see how it works from the tests now. The name is still a bit confusing to me, though. Maybe `ResourceSpecTypeMapper` or at least `ResourceSpecIdentifier`, since what it returns is a `ResourceSpec` sub-class?

Reply:
I'm going to change it to `ResourceSpecIdentifier`. It also really does not need to be an interface any more. Changing to a `@Component` class where the `SupportedKind` instances just get injected.